src/figuresClasses.ts

```diff
@@ -1,19 +1,101 @@
-export interface Figure {
+const ERROR_MESSAGE = 'Value cannot be <= 0!';
+
+function circledNumber(number: number): number {
+  return Math.floor(number * 100) / 100;
+}
 
+enum Shape {
+  Triangle = 'triangle',
+  Circle = 'circle',
+  Rectangle = 'rectangle'
+}
+enum Color {
+  Red = 'red',
+  Green = 'green',
+  Blue = 'blue'
 }
+export interface Figure {
+  shape: Shape,
+  color: Color,
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  public shape: Shape = Shape.Triangle;
 
-export class Triangle {
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (this.a <= 0 || this.b <= 0 || this.c <= 0) {
+      throw new Error(ERROR_MESSAGE);
+    }
 
+    if (this.a + this.b <= this.c
+      || this.b + this.c <= this.a
+      || this.c + this.a <= this.b) {
+      throw new Error(
+        `sides ${this.a}, ${this.b} and ${this.c} can't form a triangle`,
+      );
+    }
```

Comment:
You don't need to point to `this` in this case as you get them from the constructor